dot/rpc/websocket.go

```diff
@@ -95,3 +190,50 @@ func (h *HTTPServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 }
+
+func (h *HTTPServer) registerSubscription(conn *websocket.Conn, reqID float64, subscriptionType int) (uint32, error) {
+	wssub := h.serverConfig.WSSubscriptions
+	if wssub == nil {
+		wssub = make(map[uint32]*WebSocketSubscription)
+	}
+	sub := uint32(len(wssub)) + 1
+	wss := &WebSocketSubscription{
+		WSConnection:     conn,
+		SubscriptionType: subscriptionType,
+	}
+	wssub[sub] = wss
+	h.serverConfig.WSSubscriptions = wssub
+	initRes := newSubscriptionResponseJSON()
+	initRes.Result = sub
+	initRes.ID = reqID
+
+	return sub, conn.WriteJSON(initRes)
+}
+
+func (h *HTTPServer) blockReceivedListener() {
+	blkRec := h.serverConfig.CoreAPI.GetBlockReceivedChannel()
+	for {
+		// receive block from BABE session
+		block, ok := <-blkRec
```

Comment:
I think if a value is read from a channel, it gets taken out of the channel, if there are 2 processes reading from this channel then only one of them will be able to see the value.  I think this will cause an issue since there is a core process also reading from the same channel.
maybe creating a separate channel for websockets would be better? or inside core, in the process that already reads from the block channel, you could send that block to here as well?

Reply:
@noot I think you can have multiple go routines listening on one channel, according to this post, and the sample code included, it works. https://stackoverflow.com/questions/15715605/multiple-goroutines-listening-on-one-channel

Reply:
yeah you can, but according to that second comment, if one goroutine reads a value from the channel, then the other goroutines won't see that value, since it's already been read. 

I'm not totally sure though, I'm going to try it out

Reply:
Ok, I see I misunderstood some of the detail in the sample code.  I'll refactor this to create a separate channel.  Thanks for clarifying.